simulation/test_swarm.cpp

```diff
@@ -558,17 +558,14 @@ TORRENT_TEST(block_uploaded_alert)
 			if (auto at = lt::alert_cast<lt::add_torrent_alert>(a))
 			{
 				// init blocks vector, MUST happen before any block_uploaded_alert alerts
-				std::shared_ptr<lt::torrent> torrent = at->handle.native_handle();
-				std::shared_ptr<const torrent_info> torrent_info = torrent->get_torrent_copy();
+				auto torrent = at->handle.native_handle();
+				auto torrent_info = &torrent->torrent_file();
```

Comment:
sorry I wasn't very clear here (and I missed that you're calling ``native_handle``). But you can get the ``torrent_info`` object by just calling: ``st->handle.torrent_file()``. There's no need to pull out the internal object.